autokeras/auto/tuner.py

```diff
@@ -1,20 +1,35 @@
+import copy
+
 import tensorflow as tf
 import kerastuner
 
+from autokeras import utils
+
 
 class AutoTuner(kerastuner.Tuner):
     """Modified KerasTuner base class to include preprocessing layers."""
 
     def run_trial(self, trial, hp, fit_args, fit_kwargs):
         """Preprocess the x and y before calling the base run_trial."""
-        fit_kwargs.update(dict(zip(tf.keras.Model.fit.__code__.co_varnames,
+        new_fit_kwargs = copy.copy(fit_kwargs)
+        new_fit_kwargs.update(dict(zip(tf.keras.Model.fit.__code__.co_varnames,
```

Comment:
This does not seem like a reliable way to obtain the names of the positional arguments of `fit`, prefer using `getfullargspec`